static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx

```diff
@@ -0,0 +1,50 @@
+import {browserHistory} from 'react-router';
+import styled from '@emotion/styled';
+import {Location} from 'history';
+
+import SearchBar from 'sentry/components/events/searchBar';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Organization} from 'sentry/types';
+import EventView from 'sentry/utils/discover/eventView';
+import {decodeScalar} from 'sentry/utils/queryString';
+
+interface SpanDetailsSearchBarProps {
+  eventView: EventView;
+  location: Location;
+  organization: Organization;
+}
+
+export default function SpanDetailsSearchBar({
+  organization,
+  eventView,
+  location,
+}: SpanDetailsSearchBarProps) {
+  const query = decodeScalar(location.query.query, '');
+
+  const handleSearchQuery = (searchQuery: string): void => {
+    browserHistory.push({
+      pathname: location.pathname,
+      query: {
+        ...location.query,
+        cursor: undefined,
+        query: String(searchQuery).trim() || undefined,
+      },
+    });
+  };
```

Comment:
Not for this PR, but we should probably just make this a util considering it's basically the same for all our search bars (+ eventView conditions + cursors).

Reply:
agreed 👍